// @aptos-labs/ts-sdk/dist/core/crypto/abstraction.js
import { AuthenticationKey } from "../authenticationKey.js";
import { Hex } from "../hex.js";
import { AccountPublicKey } from "./publicKey.js";
import { Signature } from "./signature.js";
export class AbstractSignature extends Signature {
    value;
    constructor(value) {
        super();
        this.value = Hex.fromHexInput(value).toUint8Array();
    }
    serialize(serializer) {
        serializer.serializeBytes(this.value);
    }
    static deserialize(deserializer) {
        return new AbstractSignature(deserializer.deserializeBytes());
    }
}
export class AbstractPublicKey extends AccountPublicKey {
    accountAddress;
    constructor(accountAddress) {
        super();
        this.accountAddress = accountAddress;
    }
    authKey() {
        return new AuthenticationKey({ data: this.accountAddress.toUint8Array() });
    }
    verifySignature(args) {
        throw new Error("This function is not implemented for AbstractPublicKey.");
    }
    async verifySignatureAsync(args) {
        throw new Error("This function is not implemented for AbstractPublicKey.");
    }
    serialize(serializer) {
        throw new Error("This function is not implemented for AbstractPublicKey.");
    }
}
