// @aptos-labs/ts-sdk/dist/internal/abstraction.js
import { TypeTagAddress, TypeTagStruct, stringStructTag, } from "../transactions/index.js";
import { generateTransaction } from "./transactionSubmission.js";
import { getFunctionParts } from "../utils/helpers.js";
export async function addAuthenticationFunctionTransaction(args) {
    const { aptosConfig, sender, authenticationFunction, options } = args;
    const { moduleAddress, moduleName, functionName } = getFunctionParts(authenticationFunction);
    return generateTransaction({
        aptosConfig,
        sender,
        data: {
            function: "0x1::account_abstraction::add_authentication_function",
            typeArguments: [],
            functionArguments: [moduleAddress, moduleName, functionName],
            abi: {
                typeParameters: [],
                parameters: [new TypeTagAddress(), new TypeTagStruct(stringStructTag()), new TypeTagStruct(stringStructTag())],
            },
        },
        options,
    });
}
export async function removeAuthenticationFunctionTransaction(args) {
    const { aptosConfig, sender, authenticationFunction, options } = args;
    const { moduleAddress, moduleName, functionName } = getFunctionParts(authenticationFunction);
    return generateTransaction({
        aptosConfig,
        sender,
        data: {
            function: "0x1::account_abstraction::remove_authentication_function",
            typeArguments: [],
            functionArguments: [moduleAddress, moduleName, functionName],
            abi: {
                typeParameters: [],
                parameters: [new TypeTagAddress(), new TypeTagStruct(stringStructTag()), new TypeTagStruct(stringStructTag())],
            },
        },
        options,
    });
}
export async function removeDispatchableAuthenticatorTransaction(args) {
    const { aptosConfig, sender, options } = args;
    return generateTransaction({
        aptosConfig,
        sender,
        data: {
            function: "0x1::account_abstraction::remove_authenticator",
            typeArguments: [],
            functionArguments: [],
            abi: { typeParameters: [], parameters: [] },
        },
        options,
    });
}
