// @aptos-labs/ts-sdk/dist/internal/view.js
// Copyright © Aptos Foundation
// SPDX-License-Identifier: Apache-2.0
import { MimeType } from "../types/index.js";
import { generateViewFunctionPayload, } from "../transactions/index.js";
import { Serializer } from "../bcs/index.js";
import { postAptosFullNode } from "../client/index.js";
export async function view(args) {
    const { aptosConfig, payload, options } = args;
    const viewFunctionPayload = await generateViewFunctionPayload({
        ...payload,
        aptosConfig,
    });
    const serializer = new Serializer();
    viewFunctionPayload.serialize(serializer);
    const bytes = serializer.toUint8Array();
    const { data } = await postAptosFullNode({
        aptosConfig,
        path: "view",
        originMethod: "view",
        contentType: MimeType.BCS_VIEW_FUNCTION,
        params: { ledger_version: options?.ledgerVersion },
        body: bytes,
    });
    return data;
}
export async function viewJson(args) {
    const { aptosConfig, payload, options } = args;
    const { data } = await postAptosFullNode({
        aptosConfig,
        originMethod: "viewJson",
        path: "view",
        params: { ledger_version: options?.ledgerVersion },
        body: {
            function: payload.function,
            type_arguments: payload.typeArguments ?? [],
            arguments: payload.functionArguments ?? [],
        },
    });
    return data;
}
