// @aptos-labs/ts-sdk/dist/types/generated/types.js
/** select columns of table "account_transactions" */
export var AccountTransactionsSelectColumn;
(function (AccountTransactionsSelectColumn) {
    /** column name */
    AccountTransactionsSelectColumn["AccountAddress"] = "account_address";
    /** column name */
    AccountTransactionsSelectColumn["TransactionVersion"] = "transaction_version";
})(AccountTransactionsSelectColumn || (AccountTransactionsSelectColumn = {}));
/** select columns of table "address_events_summary" */
export var AddressEventsSummarySelectColumn;
(function (AddressEventsSummarySelectColumn) {
    /** column name */
    AddressEventsSummarySelectColumn["AccountAddress"] = "account_address";
    /** column name */
    AddressEventsSummarySelectColumn["MinBlockHeight"] = "min_block_height";
    /** column name */
    AddressEventsSummarySelectColumn["NumDistinctVersions"] = "num_distinct_versions";
})(AddressEventsSummarySelectColumn || (AddressEventsSummarySelectColumn = {}));
/** select columns of table "address_version_from_events" */
export var AddressVersionFromEventsSelectColumn;
(function (AddressVersionFromEventsSelectColumn) {
    /** column name */
    AddressVersionFromEventsSelectColumn["AccountAddress"] = "account_address";
    /** column name */
    AddressVersionFromEventsSelectColumn["TransactionVersion"] = "transaction_version";
})(AddressVersionFromEventsSelectColumn || (AddressVersionFromEventsSelectColumn = {}));
/** select columns of table "legacy_migration_v1.address_version_from_move_resources" */
export var AddressVersionFromMoveResourcesSelectColumn;
(function (AddressVersionFromMoveResourcesSelectColumn) {
    /** column name */
    AddressVersionFromMoveResourcesSelectColumn["Address"] = "address";
    /** column name */
    AddressVersionFromMoveResourcesSelectColumn["TransactionVersion"] = "transaction_version";
})(AddressVersionFromMoveResourcesSelectColumn || (AddressVersionFromMoveResourcesSelectColumn = {}));
/** select columns of table "auth_key_account_addresses" */
export var AuthKeyAccountAddressesSelectColumn;
(function (AuthKeyAccountAddressesSelectColumn) {
    /** column name */
    AuthKeyAccountAddressesSelectColumn["AccountAddress"] = "account_address";
    /** column name */
    AuthKeyAccountAddressesSelectColumn["AuthKey"] = "auth_key";
    /** column name */
    AuthKeyAccountAddressesSelectColumn["IsAuthKeyUsed"] = "is_auth_key_used";
    /** column name */
    AuthKeyAccountAddressesSelectColumn["LastTransactionVersion"] = "last_transaction_version";
})(AuthKeyAccountAddressesSelectColumn || (AuthKeyAccountAddressesSelectColumn = {}));
/** select columns of table "block_metadata_transactions" */
export var BlockMetadataTransactionsSelectColumn;
(function (BlockMetadataTransactionsSelectColumn) {
    /** column name */
    BlockMetadataTransactionsSelectColumn["BlockHeight"] = "block_height";
    /** column name */
    BlockMetadataTransactionsSelectColumn["Epoch"] = "epoch";
    /** column name */
    BlockMetadataTransactionsSelectColumn["FailedProposerIndices"] = "failed_proposer_indices";
    /** column name */
    BlockMetadataTransactionsSelectColumn["Id"] = "id";
    /** column name */
    BlockMetadataTransactionsSelectColumn["PreviousBlockVotesBitvec"] = "previous_block_votes_bitvec";
    /** column name */
    BlockMetadataTransactionsSelectColumn["Proposer"] = "proposer";
    /** column name */
    BlockMetadataTransactionsSelectColumn["Round"] = "round";
    /** column name */
    BlockMetadataTransactionsSelectColumn["Timestamp"] = "timestamp";
    /** column name */
    BlockMetadataTransactionsSelectColumn["Version"] = "version";
})(BlockMetadataTransactionsSelectColumn || (BlockMetadataTransactionsSelectColumn = {}));
/** select columns of table "legacy_migration_v1.coin_activities" */
export var CoinActivitiesSelectColumn;
(function (CoinActivitiesSelectColumn) {
    /** column name */
    CoinActivitiesSelectColumn["ActivityType"] = "activity_type";
    /** column name */
    CoinActivitiesSelectColumn["Amount"] = "amount";
    /** column name */
    CoinActivitiesSelectColumn["BlockHeight"] = "block_height";
    /** column name */
    CoinActivitiesSelectColumn["CoinType"] = "coin_type";
    /** column name */
    CoinActivitiesSelectColumn["EntryFunctionIdStr"] = "entry_function_id_str";
    /** column name */
    CoinActivitiesSelectColumn["EventAccountAddress"] = "event_account_address";
    /** column name */
    CoinActivitiesSelectColumn["EventCreationNumber"] = "event_creation_number";
    /** column name */
    CoinActivitiesSelectColumn["EventIndex"] = "event_index";
    /** column name */
    CoinActivitiesSelectColumn["EventSequenceNumber"] = "event_sequence_number";
    /** column name */
    CoinActivitiesSelectColumn["IsGasFee"] = "is_gas_fee";
    /** column name */
    CoinActivitiesSelectColumn["IsTransactionSuccess"] = "is_transaction_success";
    /** column name */
    CoinActivitiesSelectColumn["OwnerAddress"] = "owner_address";
    /** column name */
    CoinActivitiesSelectColumn["StorageRefundAmount"] = "storage_refund_amount";
    /** column name */
    CoinActivitiesSelectColumn["TransactionTimestamp"] = "transaction_timestamp";
    /** column name */
    CoinActivitiesSelectColumn["TransactionVersion"] = "transaction_version";
})(CoinActivitiesSelectColumn || (CoinActivitiesSelectColumn = {}));
/** select "coin_activities_aggregate_bool_exp_bool_and_arguments_columns" columns of table "legacy_migration_v1.coin_activities" */
export var CoinActivitiesSelectColumnCoinActivitiesAggregateBoolExpBoolAndArgumentsColumns;
(function (CoinActivitiesSelectColumnCoinActivitiesAggregateBoolExpBoolAndArgumentsColumns) {
    /** column name */
    CoinActivitiesSelectColumnCoinActivitiesAggregateBoolExpBoolAndArgumentsColumns["IsGasFee"] = "is_gas_fee";
    /** column name */
    CoinActivitiesSelectColumnCoinActivitiesAggregateBoolExpBoolAndArgumentsColumns["IsTransactionSuccess"] = "is_transaction_success";
})(CoinActivitiesSelectColumnCoinActivitiesAggregateBoolExpBoolAndArgumentsColumns || (CoinActivitiesSelectColumnCoinActivitiesAggregateBoolExpBoolAndArgumentsColumns = {}));
/** select "coin_activities_aggregate_bool_exp_bool_or_arguments_columns" columns of table "legacy_migration_v1.coin_activities" */
export var CoinActivitiesSelectColumnCoinActivitiesAggregateBoolExpBoolOrArgumentsColumns;
(function (CoinActivitiesSelectColumnCoinActivitiesAggregateBoolExpBoolOrArgumentsColumns) {
    /** column name */
    CoinActivitiesSelectColumnCoinActivitiesAggregateBoolExpBoolOrArgumentsColumns["IsGasFee"] = "is_gas_fee";
    /** column name */
    CoinActivitiesSelectColumnCoinActivitiesAggregateBoolExpBoolOrArgumentsColumns["IsTransactionSuccess"] = "is_transaction_success";
})(CoinActivitiesSelectColumnCoinActivitiesAggregateBoolExpBoolOrArgumentsColumns || (CoinActivitiesSelectColumnCoinActivitiesAggregateBoolExpBoolOrArgumentsColumns = {}));
/** select columns of table "legacy_migration_v1.coin_balances" */
export var CoinBalancesSelectColumn;
(function (CoinBalancesSelectColumn) {
    /** column name */
    CoinBalancesSelectColumn["Amount"] = "amount";
    /** column name */
    CoinBalancesSelectColumn["CoinType"] = "coin_type";
    /** column name */
    CoinBalancesSelectColumn["CoinTypeHash"] = "coin_type_hash";
    /** column name */
    CoinBalancesSelectColumn["OwnerAddress"] = "owner_address";
    /** column name */
    CoinBalancesSelectColumn["TransactionTimestamp"] = "transaction_timestamp";
    /** column name */
    CoinBalancesSelectColumn["TransactionVersion"] = "transaction_version";
})(CoinBalancesSelectColumn || (CoinBalancesSelectColumn = {}));
/** select columns of table "legacy_migration_v1.coin_infos" */
export var CoinInfosSelectColumn;
(function (CoinInfosSelectColumn) {
    /** column name */
    CoinInfosSelectColumn["CoinType"] = "coin_type";
    /** column name */
    CoinInfosSelectColumn["CoinTypeHash"] = "coin_type_hash";
    /** column name */
    CoinInfosSelectColumn["CreatorAddress"] = "creator_address";
    /** column name */
    CoinInfosSelectColumn["Decimals"] = "decimals";
    /** column name */
    CoinInfosSelectColumn["Name"] = "name";
    /** column name */
    CoinInfosSelectColumn["SupplyAggregatorTableHandle"] = "supply_aggregator_table_handle";
    /** column name */
    CoinInfosSelectColumn["SupplyAggregatorTableKey"] = "supply_aggregator_table_key";
    /** column name */
    CoinInfosSelectColumn["Symbol"] = "symbol";
    /** column name */
    CoinInfosSelectColumn["TransactionCreatedTimestamp"] = "transaction_created_timestamp";
    /** column name */
    CoinInfosSelectColumn["TransactionVersionCreated"] = "transaction_version_created";
})(CoinInfosSelectColumn || (CoinInfosSelectColumn = {}));
/** select columns of table "coin_supply" */
export var CoinSupplySelectColumn;
(function (CoinSupplySelectColumn) {
    /** column name */
    CoinSupplySelectColumn["CoinType"] = "coin_type";
    /** column name */
    CoinSupplySelectColumn["CoinTypeHash"] = "coin_type_hash";
    /** column name */
    CoinSupplySelectColumn["Supply"] = "supply";
    /** column name */
    CoinSupplySelectColumn["TransactionEpoch"] = "transaction_epoch";
    /** column name */
    CoinSupplySelectColumn["TransactionTimestamp"] = "transaction_timestamp";
    /** column name */
    CoinSupplySelectColumn["TransactionVersion"] = "transaction_version";
})(CoinSupplySelectColumn || (CoinSupplySelectColumn = {}));
/** select columns of table "legacy_migration_v1.collection_datas" */
export var CollectionDatasSelectColumn;
(function (CollectionDatasSelectColumn) {
    /** column name */
    CollectionDatasSelectColumn["CollectionDataIdHash"] = "collection_data_id_hash";
    /** column name */
    CollectionDatasSelectColumn["CollectionName"] = "collection_name";
    /** column name */
    CollectionDatasSelectColumn["CreatorAddress"] = "creator_address";
    /** column name */
    CollectionDatasSelectColumn["Description"] = "description";
    /** column name */
    CollectionDatasSelectColumn["DescriptionMutable"] = "description_mutable";
    /** column name */
    CollectionDatasSelectColumn["Maximum"] = "maximum";
    /** column name */
    CollectionDatasSelectColumn["MaximumMutable"] = "maximum_mutable";
    /** column name */
    CollectionDatasSelectColumn["MetadataUri"] = "metadata_uri";
    /** column name */
    CollectionDatasSelectColumn["Supply"] = "supply";
    /** column name */
    CollectionDatasSelectColumn["TableHandle"] = "table_handle";
    /** column name */
    CollectionDatasSelectColumn["TransactionTimestamp"] = "transaction_timestamp";
    /** column name */
    CollectionDatasSelectColumn["TransactionVersion"] = "transaction_version";
    /** column name */
    CollectionDatasSelectColumn["UriMutable"] = "uri_mutable";
})(CollectionDatasSelectColumn || (CollectionDatasSelectColumn = {}));
/** select columns of table "legacy_migration_v1.current_ans_lookup" */
export var CurrentAnsLookupSelectColumn;
(function (CurrentAnsLookupSelectColumn) {
    /** column name */
    CurrentAnsLookupSelectColumn["Domain"] = "domain";
    /** column name */
    CurrentAnsLookupSelectColumn["ExpirationTimestamp"] = "expiration_timestamp";
    /** column name */
    CurrentAnsLookupSelectColumn["IsDeleted"] = "is_deleted";
    /** column name */
    CurrentAnsLookupSelectColumn["LastTransactionVersion"] = "last_transaction_version";
    /** column name */
    CurrentAnsLookupSelectColumn["RegisteredAddress"] = "registered_address";
    /** column name */
    CurrentAnsLookupSelectColumn["Subdomain"] = "subdomain";
    /** column name */
    CurrentAnsLookupSelectColumn["TokenName"] = "token_name";
})(CurrentAnsLookupSelectColumn || (CurrentAnsLookupSelectColumn = {}));
/** select columns of table "current_ans_lookup_v2" */
export var CurrentAnsLookupV2SelectColumn;
(function (CurrentAnsLookupV2SelectColumn) {
    /** column name */
    CurrentAnsLookupV2SelectColumn["Domain"] = "domain";
    /** column name */
    CurrentAnsLookupV2SelectColumn["ExpirationTimestamp"] = "expiration_timestamp";
    /** column name */
    CurrentAnsLookupV2SelectColumn["IsDeleted"] = "is_deleted";
    /** column name */
    CurrentAnsLookupV2SelectColumn["LastTransactionVersion"] = "last_transaction_version";
    /** column name */
    CurrentAnsLookupV2SelectColumn["RegisteredAddress"] = "registered_address";
    /** column name */
    CurrentAnsLookupV2SelectColumn["Subdomain"] = "subdomain";
    /** column name */
    CurrentAnsLookupV2SelectColumn["TokenName"] = "token_name";
    /** column name */
    CurrentAnsLookupV2SelectColumn["TokenStandard"] = "token_standard";
})(CurrentAnsLookupV2SelectColumn || (CurrentAnsLookupV2SelectColumn = {}));
/** select columns of table "current_aptos_names" */
export var CurrentAptosNamesSelectColumn;
(function (CurrentAptosNamesSelectColumn) {
    /** column name */
    CurrentAptosNamesSelectColumn["Domain"] = "domain";
    /** column name */
    CurrentAptosNamesSelectColumn["DomainExpirationTimestamp"] = "domain_expiration_timestamp";
    /** column name */
    CurrentAptosNamesSelectColumn["DomainWithSuffix"] = "domain_with_suffix";
    /** column name */
    CurrentAptosNamesSelectColumn["ExpirationTimestamp"] = "expiration_timestamp";
    /** column name */
    CurrentAptosNamesSelectColumn["IsActive"] = "is_active";
    /** column name */
    CurrentAptosNamesSelectColumn["IsPrimary"] = "is_primary";
    /** column name */
    CurrentAptosNamesSelectColumn["LastTransactionVersion"] = "last_transaction_version";
    /** column name */
    CurrentAptosNamesSelectColumn["OwnerAddress"] = "owner_address";
    /** column name */
    CurrentAptosNamesSelectColumn["RegisteredAddress"] = "registered_address";
    /** column name */
    CurrentAptosNamesSelectColumn["Subdomain"] = "subdomain";
    /** column name */
    CurrentAptosNamesSelectColumn["SubdomainExpirationPolicy"] = "subdomain_expiration_policy";
    /** column name */
    CurrentAptosNamesSelectColumn["TokenName"] = "token_name";
    /** column name */
    CurrentAptosNamesSelectColumn["TokenStandard"] = "token_standard";
})(CurrentAptosNamesSelectColumn || (CurrentAptosNamesSelectColumn = {}));
/** select "current_aptos_names_aggregate_bool_exp_bool_and_arguments_columns" columns of table "current_aptos_names" */
export var CurrentAptosNamesSelectColumnCurrentAptosNamesAggregateBoolExpBoolAndArgumentsColumns;
(function (CurrentAptosNamesSelectColumnCurrentAptosNamesAggregateBoolExpBoolAndArgumentsColumns) {
    /** column name */
    CurrentAptosNamesSelectColumnCurrentAptosNamesAggregateBoolExpBoolAndArgumentsColumns["IsActive"] = "is_active";
    /** column name */
    CurrentAptosNamesSelectColumnCurrentAptosNamesAggregateBoolExpBoolAndArgumentsColumns["IsPrimary"] = "is_primary";
})(CurrentAptosNamesSelectColumnCurrentAptosNamesAggregateBoolExpBoolAndArgumentsColumns || (CurrentAptosNamesSelectColumnCurrentAptosNamesAggregateBoolExpBoolAndArgumentsColumns = {}));
/** select "current_aptos_names_aggregate_bool_exp_bool_or_arguments_columns" columns of table "current_aptos_names" */
export var CurrentAptosNamesSelectColumnCurrentAptosNamesAggregateBoolExpBoolOrArgumentsColumns;
(function (CurrentAptosNamesSelectColumnCurrentAptosNamesAggregateBoolExpBoolOrArgumentsColumns) {
    /** column name */
    CurrentAptosNamesSelectColumnCurrentAptosNamesAggregateBoolExpBoolOrArgumentsColumns["IsActive"] = "is_active";
    /** column name */
    CurrentAptosNamesSelectColumnCurrentAptosNamesAggregateBoolExpBoolOrArgumentsColumns["IsPrimary"] = "is_primary";
})(CurrentAptosNamesSelectColumnCurrentAptosNamesAggregateBoolExpBoolOrArgumentsColumns || (CurrentAptosNamesSelectColumnCurrentAptosNamesAggregateBoolExpBoolOrArgumentsColumns = {}));
/** select columns of table "legacy_migration_v1.current_coin_balances" */
export var CurrentCoinBalancesSelectColumn;
(function (CurrentCoinBalancesSelectColumn) {
    /** column name */
    CurrentCoinBalancesSelectColumn["Amount"] = "amount";
    /** column name */
    CurrentCoinBalancesSelectColumn["CoinType"] = "coin_type";
    /** column name */
    CurrentCoinBalancesSelectColumn["CoinTypeHash"] = "coin_type_hash";
    /** column name */
    CurrentCoinBalancesSelectColumn["LastTransactionTimestamp"] = "last_transaction_timestamp";
    /** column name */
    CurrentCoinBalancesSelectColumn["LastTransactionVersion"] = "last_transaction_version";
    /** column name */
    CurrentCoinBalancesSelectColumn["OwnerAddress"] = "owner_address";
})(CurrentCoinBalancesSelectColumn || (CurrentCoinBalancesSelectColumn = {}));
/** select columns of table "legacy_migration_v1.current_collection_datas" */
export var CurrentCollectionDatasSelectColumn;
(function (CurrentCollectionDatasSelectColumn) {
    /** column name */
    CurrentCollectionDatasSelectColumn["CollectionDataIdHash"] = "collection_data_id_hash";
    /** column name */
    CurrentCollectionDatasSelectColumn["CollectionName"] = "collection_name";
    /** column name */
    CurrentCollectionDatasSelectColumn["CreatorAddress"] = "creator_address";
    /** column name */
    CurrentCollectionDatasSelectColumn["Description"] = "description";
    /** column name */
    CurrentCollectionDatasSelectColumn["DescriptionMutable"] = "description_mutable";
    /** column name */
    CurrentCollectionDatasSelectColumn["LastTransactionTimestamp"] = "last_transaction_timestamp";
    /** column name */
    CurrentCollectionDatasSelectColumn["LastTransactionVersion"] = "last_transaction_version";
    /** column name */
    CurrentCollectionDatasSelectColumn["Maximum"] = "maximum";
    /** column name */
    CurrentCollectionDatasSelectColumn["MaximumMutable"] = "maximum_mutable";
    /** column name */
    CurrentCollectionDatasSelectColumn["MetadataUri"] = "metadata_uri";
    /** column name */
    CurrentCollectionDatasSelectColumn["Supply"] = "supply";
    /** column name */
    CurrentCollectionDatasSelectColumn["TableHandle"] = "table_handle";
    /** column name */
    CurrentCollectionDatasSelectColumn["UriMutable"] = "uri_mutable";
})(CurrentCollectionDatasSelectColumn || (CurrentCollectionDatasSelectColumn = {}));
/** select columns of table "current_collection_ownership_v2_view" */
export var CurrentCollectionOwnershipV2ViewSelectColumn;
(function (CurrentCollectionOwnershipV2ViewSelectColumn) {
    /** column name */
    CurrentCollectionOwnershipV2ViewSelectColumn["CollectionId"] = "collection_id";
    /** column name */
    CurrentCollectionOwnershipV2ViewSelectColumn["CollectionName"] = "collection_name";
    /** column name */
    CurrentCollectionOwnershipV2ViewSelectColumn["CollectionUri"] = "collection_uri";
    /** column name */
    CurrentCollectionOwnershipV2ViewSelectColumn["CreatorAddress"] = "creator_address";
    /** column name */
    CurrentCollectionOwnershipV2ViewSelectColumn["DistinctTokens"] = "distinct_tokens";
    /** column name */
    CurrentCollectionOwnershipV2ViewSelectColumn["LastTransactionVersion"] = "last_transaction_version";
    /** column name */
    CurrentCollectionOwnershipV2ViewSelectColumn["OwnerAddress"] = "owner_address";
    /** column name */
    CurrentCollectionOwnershipV2ViewSelectColumn["SingleTokenUri"] = "single_token_uri";
})(CurrentCollectionOwnershipV2ViewSelectColumn || (CurrentCollectionOwnershipV2ViewSelectColumn = {}));
/** select columns of table "current_collections_v2" */
export var CurrentCollectionsV2SelectColumn;
(function (CurrentCollectionsV2SelectColumn) {
    /** column name */
    CurrentCollectionsV2SelectColumn["CollectionId"] = "collection_id";
    /** column name */
    CurrentCollectionsV2SelectColumn["CollectionName"] = "collection_name";
    /** column name */
    CurrentCollectionsV2SelectColumn["CollectionProperties"] = "collection_properties";
    /** column name */
    CurrentCollectionsV2SelectColumn["CreatorAddress"] = "creator_address";
    /** column name */
    CurrentCollectionsV2SelectColumn["CurrentSupply"] = "current_supply";
    /** column name */
    CurrentCollectionsV2SelectColumn["Description"] = "description";
    /** column name */
    CurrentCollectionsV2SelectColumn["LastTransactionTimestamp"] = "last_transaction_timestamp";
    /** column name */
    CurrentCollectionsV2SelectColumn["LastTransactionVersion"] = "last_transaction_version";
    /** column name */
    CurrentCollectionsV2SelectColumn["MaxSupply"] = "max_supply";
    /** column name */
    CurrentCollectionsV2SelectColumn["MutableDescription"] = "mutable_description";
    /** column name */
    CurrentCollectionsV2SelectColumn["MutableUri"] = "mutable_uri";
    /** column name */
    CurrentCollectionsV2SelectColumn["TableHandleV1"] = "table_handle_v1";
    /** column name */
    CurrentCollectionsV2SelectColumn["TokenStandard"] = "token_standard";
    /** column name */
    CurrentCollectionsV2SelectColumn["TotalMintedV2"] = "total_minted_v2";
    /** column name */
    CurrentCollectionsV2SelectColumn["Uri"] = "uri";
})(CurrentCollectionsV2SelectColumn || (CurrentCollectionsV2SelectColumn = {}));
/** select columns of table "current_delegated_staking_pool_balances" */
export var CurrentDelegatedStakingPoolBalancesSelectColumn;
(function (CurrentDelegatedStakingPoolBalancesSelectColumn) {
    /** column name */
    CurrentDelegatedStakingPoolBalancesSelectColumn["ActiveTableHandle"] = "active_table_handle";
    /** column name */
    CurrentDelegatedStakingPoolBalancesSelectColumn["InactiveTableHandle"] = "inactive_table_handle";
    /** column name */
    CurrentDelegatedStakingPoolBalancesSelectColumn["LastTransactionVersion"] = "last_transaction_version";
    /** column name */
    CurrentDelegatedStakingPoolBalancesSelectColumn["OperatorCommissionPercentage"] = "operator_commission_percentage";
    /** column name */
    CurrentDelegatedStakingPoolBalancesSelectColumn["StakingPoolAddress"] = "staking_pool_address";
    /** column name */
    CurrentDelegatedStakingPoolBalancesSelectColumn["TotalCoins"] = "total_coins";
    /** column name */
    CurrentDelegatedStakingPoolBalancesSelectColumn["TotalShares"] = "total_shares";
})(CurrentDelegatedStakingPoolBalancesSelectColumn || (CurrentDelegatedStakingPoolBalancesSelectColumn = {}));
/** select columns of table "current_delegated_voter" */
export var CurrentDelegatedVoterSelectColumn;
(function (CurrentDelegatedVoterSelectColumn) {
    /** column name */
    CurrentDelegatedVoterSelectColumn["DelegationPoolAddress"] = "delegation_pool_address";
    /** column name */
    CurrentDelegatedVoterSelectColumn["DelegatorAddress"] = "delegator_address";
    /** column name */
    CurrentDelegatedVoterSelectColumn["LastTransactionTimestamp"] = "last_transaction_timestamp";
    /** column name */
    CurrentDelegatedVoterSelectColumn["LastTransactionVersion"] = "last_transaction_version";
    /** column name */
    CurrentDelegatedVoterSelectColumn["PendingVoter"] = "pending_voter";
    /** column name */
    CurrentDelegatedVoterSelectColumn["TableHandle"] = "table_handle";
    /** column name */
    CurrentDelegatedVoterSelectColumn["Voter"] = "voter";
})(CurrentDelegatedVoterSelectColumn || (CurrentDelegatedVoterSelectColumn = {}));
/** select columns of table "current_delegator_balances" */
export var CurrentDelegatorBalancesSelectColumn;
(function (CurrentDelegatorBalancesSelectColumn) {
    /** column name */
    CurrentDelegatorBalancesSelectColumn["DelegatorAddress"] = "delegator_address";
    /** column name */
    CurrentDelegatorBalancesSelectColumn["LastTransactionVersion"] = "last_transaction_version";
    /** column name */
    CurrentDelegatorBalancesSelectColumn["ParentTableHandle"] = "parent_table_handle";
    /** column name */
    CurrentDelegatorBalancesSelectColumn["PoolAddress"] = "pool_address";
    /** column name */
    CurrentDelegatorBalancesSelectColumn["PoolType"] = "pool_type";
    /** column name */
    CurrentDelegatorBalancesSelectColumn["Shares"] = "shares";
    /** column name */
    CurrentDelegatorBalancesSelectColumn["TableHandle"] = "table_handle";
})(CurrentDelegatorBalancesSelectColumn || (CurrentDelegatorBalancesSelectColumn = {}));
/** select columns of table "current_fungible_asset_balances" */
export var CurrentFungibleAssetBalancesSelectColumn;
(function (CurrentFungibleAssetBalancesSelectColumn) {
    /** column name */
    CurrentFungibleAssetBalancesSelectColumn["Amount"] = "amount";
    /** column name */
    CurrentFungibleAssetBalancesSelectColumn["AmountV1"] = "amount_v1";
    /** column name */
    CurrentFungibleAssetBalancesSelectColumn["AmountV2"] = "amount_v2";
    /** column name */
    CurrentFungibleAssetBalancesSelectColumn["AssetType"] = "asset_type";
    /** column name */
    CurrentFungibleAssetBalancesSelectColumn["AssetTypeV1"] = "asset_type_v1";
    /** column name */
    CurrentFungibleAssetBalancesSelectColumn["AssetTypeV2"] = "asset_type_v2";
    /** column name */
    CurrentFungibleAssetBalancesSelectColumn["IsFrozen"] = "is_frozen";
    /** column name */
    CurrentFungibleAssetBalancesSelectColumn["IsPrimary"] = "is_primary";
    /** column name */
    CurrentFungibleAssetBalancesSelectColumn["LastTransactionTimestamp"] = "last_transaction_timestamp";
    /** column name */
    CurrentFungibleAssetBalancesSelectColumn["LastTransactionTimestampV1"] = "last_transaction_timestamp_v1";
    /** column name */
    CurrentFungibleAssetBalancesSelectColumn["LastTransactionTimestampV2"] = "last_transaction_timestamp_v2";
    /** column name */
    CurrentFungibleAssetBalancesSelectColumn["LastTransactionVersion"] = "last_transaction_version";
    /** column name */
    CurrentFungibleAssetBalancesSelectColumn["LastTransactionVersionV1"] = "last_transaction_version_v1";
    /** column name */
    CurrentFungibleAssetBalancesSelectColumn["LastTransactionVersionV2"] = "last_transaction_version_v2";
    /** column name */
    CurrentFungibleAssetBalancesSelectColumn["OwnerAddress"] = "owner_address";
    /** column name */
    CurrentFungibleAssetBalancesSelectColumn["StorageId"] = "storage_id";
    /** column name */
    CurrentFungibleAssetBalancesSelectColumn["TokenStandard"] = "token_standard";
})(CurrentFungibleAssetBalancesSelectColumn || (CurrentFungibleAssetBalancesSelectColumn = {}));
/** select columns of table "current_objects" */
export var CurrentObjectsSelectColumn;
(function (CurrentObjectsSelectColumn) {
    /** column name */
    CurrentObjectsSelectColumn["AllowUngatedTransfer"] = "allow_ungated_transfer";
    /** column name */
    CurrentObjectsSelectColumn["IsDeleted"] = "is_deleted";
    /** column name */
    CurrentObjectsSelectColumn["LastGuidCreationNum"] = "last_guid_creation_num";
    /** column name */
    CurrentObjectsSelectColumn["LastTransactionVersion"] = "last_transaction_version";
    /** column name */
    CurrentObjectsSelectColumn["ObjectAddress"] = "object_address";
    /** column name */
    CurrentObjectsSelectColumn["OwnerAddress"] = "owner_address";
    /** column name */
    CurrentObjectsSelectColumn["StateKeyHash"] = "state_key_hash";
})(CurrentObjectsSelectColumn || (CurrentObjectsSelectColumn = {}));
/** select columns of table "current_staking_pool_voter" */
export var CurrentStakingPoolVoterSelectColumn;
(function (CurrentStakingPoolVoterSelectColumn) {
    /** column name */
    CurrentStakingPoolVoterSelectColumn["LastTransactionVersion"] = "last_transaction_version";
    /** column name */
    CurrentStakingPoolVoterSelectColumn["OperatorAddress"] = "operator_address";
    /** column name */
    CurrentStakingPoolVoterSelectColumn["StakingPoolAddress"] = "staking_pool_address";
    /** column name */
    CurrentStakingPoolVoterSelectColumn["VoterAddress"] = "voter_address";
})(CurrentStakingPoolVoterSelectColumn || (CurrentStakingPoolVoterSelectColumn = {}));
/** select columns of table "current_table_items" */
export var CurrentTableItemsSelectColumn;
(function (CurrentTableItemsSelectColumn) {
    /** column name */
    CurrentTableItemsSelectColumn["DecodedKey"] = "decoded_key";
    /** column name */
    CurrentTableItemsSelectColumn["DecodedValue"] = "decoded_value";
    /** column name */
    CurrentTableItemsSelectColumn["IsDeleted"] = "is_deleted";
    /** column name */
    CurrentTableItemsSelectColumn["Key"] = "key";
    /** column name */
    CurrentTableItemsSelectColumn["KeyHash"] = "key_hash";
    /** column name */
    CurrentTableItemsSelectColumn["LastTransactionVersion"] = "last_transaction_version";
    /** column name */
    CurrentTableItemsSelectColumn["TableHandle"] = "table_handle";
})(CurrentTableItemsSelectColumn || (CurrentTableItemsSelectColumn = {}));
/** select columns of table "legacy_migration_v1.current_token_datas" */
export var CurrentTokenDatasSelectColumn;
(function (CurrentTokenDatasSelectColumn) {
    /** column name */
    CurrentTokenDatasSelectColumn["CollectionDataIdHash"] = "collection_data_id_hash";
    /** column name */
    CurrentTokenDatasSelectColumn["CollectionName"] = "collection_name";
    /** column name */
    CurrentTokenDatasSelectColumn["CreatorAddress"] = "creator_address";
    /** column name */
    CurrentTokenDatasSelectColumn["DefaultProperties"] = "default_properties";
    /** column name */
    CurrentTokenDatasSelectColumn["Description"] = "description";
    /** column name */
    CurrentTokenDatasSelectColumn["DescriptionMutable"] = "description_mutable";
    /** column name */
    CurrentTokenDatasSelectColumn["LargestPropertyVersion"] = "largest_property_version";
    /** column name */
    CurrentTokenDatasSelectColumn["LastTransactionTimestamp"] = "last_transaction_timestamp";
    /** column name */
    CurrentTokenDatasSelectColumn["LastTransactionVersion"] = "last_transaction_version";
    /** column name */
    CurrentTokenDatasSelectColumn["Maximum"] = "maximum";
    /** column name */
    CurrentTokenDatasSelectColumn["MaximumMutable"] = "maximum_mutable";
    /** column name */
    CurrentTokenDatasSelectColumn["MetadataUri"] = "metadata_uri";
    /** column name */
    CurrentTokenDatasSelectColumn["Name"] = "name";
    /** column name */
    CurrentTokenDatasSelectColumn["PayeeAddress"] = "payee_address";
    /** column name */
    CurrentTokenDatasSelectColumn["PropertiesMutable"] = "properties_mutable";
    /** column name */
    CurrentTokenDatasSelectColumn["RoyaltyMutable"] = "royalty_mutable";
    /** column name */
    CurrentTokenDatasSelectColumn["RoyaltyPointsDenominator"] = "royalty_points_denominator";
    /** column name */
    CurrentTokenDatasSelectColumn["RoyaltyPointsNumerator"] = "royalty_points_numerator";
    /** column name */
    CurrentTokenDatasSelectColumn["Supply"] = "supply";
    /** column name */
    CurrentTokenDatasSelectColumn["TokenDataIdHash"] = "token_data_id_hash";
    /** column name */
    CurrentTokenDatasSelectColumn["UriMutable"] = "uri_mutable";
})(CurrentTokenDatasSelectColumn || (CurrentTokenDatasSelectColumn = {}));
/** select columns of table "current_token_datas_v2" */
export var CurrentTokenDatasV2SelectColumn;
(function (CurrentTokenDatasV2SelectColumn) {
    /** column name */
    CurrentTokenDatasV2SelectColumn["CollectionId"] = "collection_id";
    /** column name */
    CurrentTokenDatasV2SelectColumn["Decimals"] = "decimals";
    /** column name */
    CurrentTokenDatasV2SelectColumn["Description"] = "description";
    /** column name */
    CurrentTokenDatasV2SelectColumn["IsDeletedV2"] = "is_deleted_v2";
    /** column name */
    CurrentTokenDatasV2SelectColumn["IsFungibleV2"] = "is_fungible_v2";
    /** column name */
    CurrentTokenDatasV2SelectColumn["LargestPropertyVersionV1"] = "largest_property_version_v1";
    /** column name */
    CurrentTokenDatasV2SelectColumn["LastTransactionTimestamp"] = "last_transaction_timestamp";
    /** column name */
    CurrentTokenDatasV2SelectColumn["LastTransactionVersion"] = "last_transaction_version";
    /** column name */
    CurrentTokenDatasV2SelectColumn["Maximum"] = "maximum";
    /** column name */
    CurrentTokenDatasV2SelectColumn["Supply"] = "supply";
    /** column name */
    CurrentTokenDatasV2SelectColumn["TokenDataId"] = "token_data_id";
    /** column name */
    CurrentTokenDatasV2SelectColumn["TokenName"] = "token_name";
    /** column name */
    CurrentTokenDatasV2SelectColumn["TokenProperties"] = "token_properties";
    /** column name */
    CurrentTokenDatasV2SelectColumn["TokenStandard"] = "token_standard";
    /** column name */
    CurrentTokenDatasV2SelectColumn["TokenUri"] = "token_uri";
})(CurrentTokenDatasV2SelectColumn || (CurrentTokenDatasV2SelectColumn = {}));
/** select columns of table "legacy_migration_v1.current_token_ownerships" */
export var CurrentTokenOwnershipsSelectColumn;
(function (CurrentTokenOwnershipsSelectColumn) {
    /** column name */
    CurrentTokenOwnershipsSelectColumn["Amount"] = "amount";
    /** column name */
    CurrentTokenOwnershipsSelectColumn["CollectionDataIdHash"] = "collection_data_id_hash";
    /** column name */
    CurrentTokenOwnershipsSelectColumn["CollectionName"] = "collection_name";
    /** column name */
    CurrentTokenOwnershipsSelectColumn["CreatorAddress"] = "creator_address";
    /** column name */
    CurrentTokenOwnershipsSelectColumn["LastTransactionTimestamp"] = "last_transaction_timestamp";
    /** column name */
    CurrentTokenOwnershipsSelectColumn["LastTransactionVersion"] = "last_transaction_version";
    /** column name */
    CurrentTokenOwnershipsSelectColumn["Name"] = "name";
    /** column name */
    CurrentTokenOwnershipsSelectColumn["OwnerAddress"] = "owner_address";
    /** column name */
    CurrentTokenOwnershipsSelectColumn["PropertyVersion"] = "property_version";
    /** column name */
    CurrentTokenOwnershipsSelectColumn["TableType"] = "table_type";
    /** column name */
    CurrentTokenOwnershipsSelectColumn["TokenDataIdHash"] = "token_data_id_hash";
    /** column name */
    CurrentTokenOwnershipsSelectColumn["TokenProperties"] = "token_properties";
})(CurrentTokenOwnershipsSelectColumn || (CurrentTokenOwnershipsSelectColumn = {}));
/** select columns of table "current_token_ownerships_v2" */
export var CurrentTokenOwnershipsV2SelectColumn;
(function (CurrentTokenOwnershipsV2SelectColumn) {
    /** column name */
    CurrentTokenOwnershipsV2SelectColumn["Amount"] = "amount";
    /** column name */
    CurrentTokenOwnershipsV2SelectColumn["IsFungibleV2"] = "is_fungible_v2";
    /** column name */
    CurrentTokenOwnershipsV2SelectColumn["IsSoulboundV2"] = "is_soulbound_v2";
    /** column name */
    CurrentTokenOwnershipsV2SelectColumn["LastTransactionTimestamp"] = "last_transaction_timestamp";
    /** column name */
    CurrentTokenOwnershipsV2SelectColumn["LastTransactionVersion"] = "last_transaction_version";
    /** column name */
    CurrentTokenOwnershipsV2SelectColumn["NonTransferrableByOwner"] = "non_transferrable_by_owner";
    /** column name */
    CurrentTokenOwnershipsV2SelectColumn["OwnerAddress"] = "owner_address";
    /** column name */
    CurrentTokenOwnershipsV2SelectColumn["PropertyVersionV1"] = "property_version_v1";
    /** column name */
    CurrentTokenOwnershipsV2SelectColumn["StorageId"] = "storage_id";
    /** column name */
    CurrentTokenOwnershipsV2SelectColumn["TableTypeV1"] = "table_type_v1";
    /** column name */
    CurrentTokenOwnershipsV2SelectColumn["TokenDataId"] = "token_data_id";
    /** column name */
    CurrentTokenOwnershipsV2SelectColumn["TokenPropertiesMutatedV1"] = "token_properties_mutated_v1";
    /** column name */
    CurrentTokenOwnershipsV2SelectColumn["TokenStandard"] = "token_standard";
})(CurrentTokenOwnershipsV2SelectColumn || (CurrentTokenOwnershipsV2SelectColumn = {}));
/** select "current_token_ownerships_v2_aggregate_bool_exp_bool_and_arguments_columns" columns of table "current_token_ownerships_v2" */
export var CurrentTokenOwnershipsV2SelectColumnCurrentTokenOwnershipsV2AggregateBoolExpBoolAndArgumentsColumns;
(function (CurrentTokenOwnershipsV2SelectColumnCurrentTokenOwnershipsV2AggregateBoolExpBoolAndArgumentsColumns) {
    /** column name */
    CurrentTokenOwnershipsV2SelectColumnCurrentTokenOwnershipsV2AggregateBoolExpBoolAndArgumentsColumns["IsFungibleV2"] = "is_fungible_v2";
    /** column name */
    CurrentTokenOwnershipsV2SelectColumnCurrentTokenOwnershipsV2AggregateBoolExpBoolAndArgumentsColumns["IsSoulboundV2"] = "is_soulbound_v2";
    /** column name */
    CurrentTokenOwnershipsV2SelectColumnCurrentTokenOwnershipsV2AggregateBoolExpBoolAndArgumentsColumns["NonTransferrableByOwner"] = "non_transferrable_by_owner";
})(CurrentTokenOwnershipsV2SelectColumnCurrentTokenOwnershipsV2AggregateBoolExpBoolAndArgumentsColumns || (CurrentTokenOwnershipsV2SelectColumnCurrentTokenOwnershipsV2AggregateBoolExpBoolAndArgumentsColumns = {}));
/** select "current_token_ownerships_v2_aggregate_bool_exp_bool_or_arguments_columns" columns of table "current_token_ownerships_v2" */
export var CurrentTokenOwnershipsV2SelectColumnCurrentTokenOwnershipsV2AggregateBoolExpBoolOrArgumentsColumns;
(function (CurrentTokenOwnershipsV2SelectColumnCurrentTokenOwnershipsV2AggregateBoolExpBoolOrArgumentsColumns) {
    /** column name */
    CurrentTokenOwnershipsV2SelectColumnCurrentTokenOwnershipsV2AggregateBoolExpBoolOrArgumentsColumns["IsFungibleV2"] = "is_fungible_v2";
    /** column name */
    CurrentTokenOwnershipsV2SelectColumnCurrentTokenOwnershipsV2AggregateBoolExpBoolOrArgumentsColumns["IsSoulboundV2"] = "is_soulbound_v2";
    /** column name */
    CurrentTokenOwnershipsV2SelectColumnCurrentTokenOwnershipsV2AggregateBoolExpBoolOrArgumentsColumns["NonTransferrableByOwner"] = "non_transferrable_by_owner";
})(CurrentTokenOwnershipsV2SelectColumnCurrentTokenOwnershipsV2AggregateBoolExpBoolOrArgumentsColumns || (CurrentTokenOwnershipsV2SelectColumnCurrentTokenOwnershipsV2AggregateBoolExpBoolOrArgumentsColumns = {}));
/** select columns of table "current_token_pending_claims" */
export var CurrentTokenPendingClaimsSelectColumn;
(function (CurrentTokenPendingClaimsSelectColumn) {
    /** column name */
    CurrentTokenPendingClaimsSelectColumn["Amount"] = "amount";
    /** column name */
    CurrentTokenPendingClaimsSelectColumn["CollectionDataIdHash"] = "collection_data_id_hash";
    /** column name */
    CurrentTokenPendingClaimsSelectColumn["CollectionId"] = "collection_id";
    /** column name */
    CurrentTokenPendingClaimsSelectColumn["CollectionName"] = "collection_name";
    /** column name */
    CurrentTokenPendingClaimsSelectColumn["CreatorAddress"] = "creator_address";
    /** column name */
    CurrentTokenPendingClaimsSelectColumn["FromAddress"] = "from_address";
    /** column name */
    CurrentTokenPendingClaimsSelectColumn["LastTransactionTimestamp"] = "last_transaction_timestamp";
    /** column name */
    CurrentTokenPendingClaimsSelectColumn["LastTransactionVersion"] = "last_transaction_version";
    /** column name */
    CurrentTokenPendingClaimsSelectColumn["Name"] = "name";
    /** column name */
    CurrentTokenPendingClaimsSelectColumn["PropertyVersion"] = "property_version";
    /** column name */
    CurrentTokenPendingClaimsSelectColumn["TableHandle"] = "table_handle";
    /** column name */
    CurrentTokenPendingClaimsSelectColumn["ToAddress"] = "to_address";
    /** column name */
    CurrentTokenPendingClaimsSelectColumn["TokenDataId"] = "token_data_id";
    /** column name */
    CurrentTokenPendingClaimsSelectColumn["TokenDataIdHash"] = "token_data_id_hash";
})(CurrentTokenPendingClaimsSelectColumn || (CurrentTokenPendingClaimsSelectColumn = {}));
/** select columns of table "current_token_royalty_v1" */
export var CurrentTokenRoyaltyV1SelectColumn;
(function (CurrentTokenRoyaltyV1SelectColumn) {
    /** column name */
    CurrentTokenRoyaltyV1SelectColumn["LastTransactionTimestamp"] = "last_transaction_timestamp";
    /** column name */
    CurrentTokenRoyaltyV1SelectColumn["LastTransactionVersion"] = "last_transaction_version";
    /** column name */
    CurrentTokenRoyaltyV1SelectColumn["PayeeAddress"] = "payee_address";
    /** column name */
    CurrentTokenRoyaltyV1SelectColumn["RoyaltyPointsDenominator"] = "royalty_points_denominator";
    /** column name */
    CurrentTokenRoyaltyV1SelectColumn["RoyaltyPointsNumerator"] = "royalty_points_numerator";
    /** column name */
    CurrentTokenRoyaltyV1SelectColumn["TokenDataId"] = "token_data_id";
})(CurrentTokenRoyaltyV1SelectColumn || (CurrentTokenRoyaltyV1SelectColumn = {}));
/** ordering argument of a cursor */
export var CursorOrdering;
(function (CursorOrdering) {
    /** ascending ordering of the cursor */
    CursorOrdering["Asc"] = "ASC";
    /** descending ordering of the cursor */
    CursorOrdering["Desc"] = "DESC";
})(CursorOrdering || (CursorOrdering = {}));
/** select columns of table "delegated_staking_activities" */
export var DelegatedStakingActivitiesSelectColumn;
(function (DelegatedStakingActivitiesSelectColumn) {
    /** column name */
    DelegatedStakingActivitiesSelectColumn["Amount"] = "amount";
    /** column name */
    DelegatedStakingActivitiesSelectColumn["DelegatorAddress"] = "delegator_address";
    /** column name */
    DelegatedStakingActivitiesSelectColumn["EventIndex"] = "event_index";
    /** column name */
    DelegatedStakingActivitiesSelectColumn["EventType"] = "event_type";
    /** column name */
    DelegatedStakingActivitiesSelectColumn["PoolAddress"] = "pool_address";
    /** column name */
    DelegatedStakingActivitiesSelectColumn["TransactionVersion"] = "transaction_version";
})(DelegatedStakingActivitiesSelectColumn || (DelegatedStakingActivitiesSelectColumn = {}));
/** select columns of table "delegated_staking_pool_balances" */
export var DelegatedStakingPoolBalancesSelectColumn;
(function (DelegatedStakingPoolBalancesSelectColumn) {
    /** column name */
    DelegatedStakingPoolBalancesSelectColumn["ActiveTableHandle"] = "active_table_handle";
    /** column name */
    DelegatedStakingPoolBalancesSelectColumn["InactiveTableHandle"] = "inactive_table_handle";
    /** column name */
    DelegatedStakingPoolBalancesSelectColumn["OperatorCommissionPercentage"] = "operator_commission_percentage";
    /** column name */
    DelegatedStakingPoolBalancesSelectColumn["StakingPoolAddress"] = "staking_pool_address";
    /** column name */
    DelegatedStakingPoolBalancesSelectColumn["TotalCoins"] = "total_coins";
    /** column name */
    DelegatedStakingPoolBalancesSelectColumn["TotalShares"] = "total_shares";
    /** column name */
    DelegatedStakingPoolBalancesSelectColumn["TransactionVersion"] = "transaction_version";
})(DelegatedStakingPoolBalancesSelectColumn || (DelegatedStakingPoolBalancesSelectColumn = {}));
/** select columns of table "delegated_staking_pools" */
export var DelegatedStakingPoolsSelectColumn;
(function (DelegatedStakingPoolsSelectColumn) {
    /** column name */
    DelegatedStakingPoolsSelectColumn["FirstTransactionVersion"] = "first_transaction_version";
    /** column name */
    DelegatedStakingPoolsSelectColumn["StakingPoolAddress"] = "staking_pool_address";
})(DelegatedStakingPoolsSelectColumn || (DelegatedStakingPoolsSelectColumn = {}));
/** select columns of table "delegator_distinct_pool" */
export var DelegatorDistinctPoolSelectColumn;
(function (DelegatorDistinctPoolSelectColumn) {
    /** column name */
    DelegatorDistinctPoolSelectColumn["DelegatorAddress"] = "delegator_address";
    /** column name */
    DelegatorDistinctPoolSelectColumn["PoolAddress"] = "pool_address";
})(DelegatorDistinctPoolSelectColumn || (DelegatorDistinctPoolSelectColumn = {}));
/** select columns of table "events" */
export var EventsSelectColumn;
(function (EventsSelectColumn) {
    /** column name */
    EventsSelectColumn["AccountAddress"] = "account_address";
    /** column name */
    EventsSelectColumn["CreationNumber"] = "creation_number";
    /** column name */
    EventsSelectColumn["Data"] = "data";
    /** column name */
    EventsSelectColumn["EventIndex"] = "event_index";
    /** column name */
    EventsSelectColumn["IndexedType"] = "indexed_type";
    /** column name */
    EventsSelectColumn["SequenceNumber"] = "sequence_number";
    /** column name */
    EventsSelectColumn["TransactionBlockHeight"] = "transaction_block_height";
    /** column name */
    EventsSelectColumn["TransactionVersion"] = "transaction_version";
    /** column name */
    EventsSelectColumn["Type"] = "type";
})(EventsSelectColumn || (EventsSelectColumn = {}));
/** select columns of table "fungible_asset_activities" */
export var FungibleAssetActivitiesSelectColumn;
(function (FungibleAssetActivitiesSelectColumn) {
    /** column name */
    FungibleAssetActivitiesSelectColumn["Amount"] = "amount";
    /** column name */
    FungibleAssetActivitiesSelectColumn["AssetType"] = "asset_type";
    /** column name */
    FungibleAssetActivitiesSelectColumn["BlockHeight"] = "block_height";
    /** column name */
    FungibleAssetActivitiesSelectColumn["EntryFunctionIdStr"] = "entry_function_id_str";
    /** column name */
    FungibleAssetActivitiesSelectColumn["EventIndex"] = "event_index";
    /** column name */
    FungibleAssetActivitiesSelectColumn["GasFeePayerAddress"] = "gas_fee_payer_address";
    /** column name */
    FungibleAssetActivitiesSelectColumn["IsFrozen"] = "is_frozen";
    /** column name */
    FungibleAssetActivitiesSelectColumn["IsGasFee"] = "is_gas_fee";
    /** column name */
    FungibleAssetActivitiesSelectColumn["IsTransactionSuccess"] = "is_transaction_success";
    /** column name */
    FungibleAssetActivitiesSelectColumn["OwnerAddress"] = "owner_address";
    /** column name */
    FungibleAssetActivitiesSelectColumn["StorageId"] = "storage_id";
    /** column name */
    FungibleAssetActivitiesSelectColumn["StorageRefundAmount"] = "storage_refund_amount";
    /** column name */
    FungibleAssetActivitiesSelectColumn["TokenStandard"] = "token_standard";
    /** column name */
    FungibleAssetActivitiesSelectColumn["TransactionTimestamp"] = "transaction_timestamp";
    /** column name */
    FungibleAssetActivitiesSelectColumn["TransactionVersion"] = "transaction_version";
    /** column name */
    FungibleAssetActivitiesSelectColumn["Type"] = "type";
})(FungibleAssetActivitiesSelectColumn || (FungibleAssetActivitiesSelectColumn = {}));
/** select columns of table "fungible_asset_metadata" */
export var FungibleAssetMetadataSelectColumn;
(function (FungibleAssetMetadataSelectColumn) {
    /** column name */
    FungibleAssetMetadataSelectColumn["AssetType"] = "asset_type";
    /** column name */
    FungibleAssetMetadataSelectColumn["CreatorAddress"] = "creator_address";
    /** column name */
    FungibleAssetMetadataSelectColumn["Decimals"] = "decimals";
    /** column name */
    FungibleAssetMetadataSelectColumn["IconUri"] = "icon_uri";
    /** column name */
    FungibleAssetMetadataSelectColumn["LastTransactionTimestamp"] = "last_transaction_timestamp";
    /** column name */
    FungibleAssetMetadataSelectColumn["LastTransactionVersion"] = "last_transaction_version";
    /** column name */
    FungibleAssetMetadataSelectColumn["MaximumV2"] = "maximum_v2";
    /** column name */
    FungibleAssetMetadataSelectColumn["Name"] = "name";
    /** column name */
    FungibleAssetMetadataSelectColumn["ProjectUri"] = "project_uri";
    /** column name */
    FungibleAssetMetadataSelectColumn["SupplyAggregatorTableHandleV1"] = "supply_aggregator_table_handle_v1";
    /** column name */
    FungibleAssetMetadataSelectColumn["SupplyAggregatorTableKeyV1"] = "supply_aggregator_table_key_v1";
    /** column name */
    FungibleAssetMetadataSelectColumn["SupplyV2"] = "supply_v2";
    /** column name */
    FungibleAssetMetadataSelectColumn["Symbol"] = "symbol";
    /** column name */
    FungibleAssetMetadataSelectColumn["TokenStandard"] = "token_standard";
})(FungibleAssetMetadataSelectColumn || (FungibleAssetMetadataSelectColumn = {}));
/** select columns of table "indexer_status" */
export var IndexerStatusSelectColumn;
(function (IndexerStatusSelectColumn) {
    /** column name */
    IndexerStatusSelectColumn["Db"] = "db";
    /** column name */
    IndexerStatusSelectColumn["IsIndexerUp"] = "is_indexer_up";
})(IndexerStatusSelectColumn || (IndexerStatusSelectColumn = {}));
/** select columns of table "processor_metadata.ledger_infos" */
export var LedgerInfosSelectColumn;
(function (LedgerInfosSelectColumn) {
    /** column name */
    LedgerInfosSelectColumn["ChainId"] = "chain_id";
})(LedgerInfosSelectColumn || (LedgerInfosSelectColumn = {}));
/** select columns of table "legacy_migration_v1.move_resources" */
export var MoveResourcesSelectColumn;
(function (MoveResourcesSelectColumn) {
    /** column name */
    MoveResourcesSelectColumn["Address"] = "address";
    /** column name */
    MoveResourcesSelectColumn["TransactionVersion"] = "transaction_version";
})(MoveResourcesSelectColumn || (MoveResourcesSelectColumn = {}));
/** select columns of table "nft_metadata_crawler.parsed_asset_uris" */
export var NftMetadataCrawlerParsedAssetUrisSelectColumn;
(function (NftMetadataCrawlerParsedAssetUrisSelectColumn) {
    /** column name */
    NftMetadataCrawlerParsedAssetUrisSelectColumn["AnimationOptimizerRetryCount"] = "animation_optimizer_retry_count";
    /** column name */
    NftMetadataCrawlerParsedAssetUrisSelectColumn["AssetUri"] = "asset_uri";
    /** column name */
    NftMetadataCrawlerParsedAssetUrisSelectColumn["CdnAnimationUri"] = "cdn_animation_uri";
    /** column name */
    NftMetadataCrawlerParsedAssetUrisSelectColumn["CdnImageUri"] = "cdn_image_uri";
    /** column name */
    NftMetadataCrawlerParsedAssetUrisSelectColumn["CdnJsonUri"] = "cdn_json_uri";
    /** column name */
    NftMetadataCrawlerParsedAssetUrisSelectColumn["ImageOptimizerRetryCount"] = "image_optimizer_retry_count";
    /** column name */
    NftMetadataCrawlerParsedAssetUrisSelectColumn["JsonParserRetryCount"] = "json_parser_retry_count";
    /** column name */
    NftMetadataCrawlerParsedAssetUrisSelectColumn["RawAnimationUri"] = "raw_animation_uri";
    /** column name */
    NftMetadataCrawlerParsedAssetUrisSelectColumn["RawImageUri"] = "raw_image_uri";
})(NftMetadataCrawlerParsedAssetUrisSelectColumn || (NftMetadataCrawlerParsedAssetUrisSelectColumn = {}));
/** select columns of table "num_active_delegator_per_pool" */
export var NumActiveDelegatorPerPoolSelectColumn;
(function (NumActiveDelegatorPerPoolSelectColumn) {
    /** column name */
    NumActiveDelegatorPerPoolSelectColumn["NumActiveDelegator"] = "num_active_delegator";
    /** column name */
    NumActiveDelegatorPerPoolSelectColumn["PoolAddress"] = "pool_address";
})(NumActiveDelegatorPerPoolSelectColumn || (NumActiveDelegatorPerPoolSelectColumn = {}));
/** column ordering options */
export var OrderBy;
(function (OrderBy) {
    /** in ascending order, nulls last */
    OrderBy["Asc"] = "asc";
    /** in ascending order, nulls first */
    OrderBy["AscNullsFirst"] = "asc_nulls_first";
    /** in ascending order, nulls last */
    OrderBy["AscNullsLast"] = "asc_nulls_last";
    /** in descending order, nulls first */
    OrderBy["Desc"] = "desc";
    /** in descending order, nulls first */
    OrderBy["DescNullsFirst"] = "desc_nulls_first";
    /** in descending order, nulls last */
    OrderBy["DescNullsLast"] = "desc_nulls_last";
})(OrderBy || (OrderBy = {}));
/** select columns of table "processor_metadata.processor_status" */
export var ProcessorStatusSelectColumn;
(function (ProcessorStatusSelectColumn) {
    /** column name */
    ProcessorStatusSelectColumn["LastSuccessVersion"] = "last_success_version";
    /** column name */
    ProcessorStatusSelectColumn["LastTransactionTimestamp"] = "last_transaction_timestamp";
    /** column name */
    ProcessorStatusSelectColumn["LastUpdated"] = "last_updated";
    /** column name */
    ProcessorStatusSelectColumn["Processor"] = "processor";
})(ProcessorStatusSelectColumn || (ProcessorStatusSelectColumn = {}));
/** select columns of table "proposal_votes" */
export var ProposalVotesSelectColumn;
(function (ProposalVotesSelectColumn) {
    /** column name */
    ProposalVotesSelectColumn["NumVotes"] = "num_votes";
    /** column name */
    ProposalVotesSelectColumn["ProposalId"] = "proposal_id";
    /** column name */
    ProposalVotesSelectColumn["ShouldPass"] = "should_pass";
    /** column name */
    ProposalVotesSelectColumn["StakingPoolAddress"] = "staking_pool_address";
    /** column name */
    ProposalVotesSelectColumn["TransactionTimestamp"] = "transaction_timestamp";
    /** column name */
    ProposalVotesSelectColumn["TransactionVersion"] = "transaction_version";
    /** column name */
    ProposalVotesSelectColumn["VoterAddress"] = "voter_address";
})(ProposalVotesSelectColumn || (ProposalVotesSelectColumn = {}));
/** select columns of table "public_key_auth_keys" */
export var PublicKeyAuthKeysSelectColumn;
(function (PublicKeyAuthKeysSelectColumn) {
    /** column name */
    PublicKeyAuthKeysSelectColumn["AccountPublicKey"] = "account_public_key";
    /** column name */
    PublicKeyAuthKeysSelectColumn["AuthKey"] = "auth_key";
    /** column name */
    PublicKeyAuthKeysSelectColumn["IsPublicKeyUsed"] = "is_public_key_used";
    /** column name */
    PublicKeyAuthKeysSelectColumn["LastTransactionVersion"] = "last_transaction_version";
    /** column name */
    PublicKeyAuthKeysSelectColumn["PublicKey"] = "public_key";
    /** column name */
    PublicKeyAuthKeysSelectColumn["PublicKeyType"] = "public_key_type";
    /** column name */
    PublicKeyAuthKeysSelectColumn["SignatureType"] = "signature_type";
})(PublicKeyAuthKeysSelectColumn || (PublicKeyAuthKeysSelectColumn = {}));
/** select columns of table "signatures" */
export var SignaturesSelectColumn;
(function (SignaturesSelectColumn) {
    /** column name */
    SignaturesSelectColumn["IsSenderPrimary"] = "is_sender_primary";
    /** column name */
    SignaturesSelectColumn["MultiAgentIndex"] = "multi_agent_index";
    /** column name */
    SignaturesSelectColumn["MultiSigIndex"] = "multi_sig_index";
    /** column name */
    SignaturesSelectColumn["PublicKey"] = "public_key";
    /** column name */
    SignaturesSelectColumn["PublicKeyIndices"] = "public_key_indices";
    /** column name */
    SignaturesSelectColumn["Signature"] = "signature";
    /** column name */
    SignaturesSelectColumn["Signer"] = "signer";
    /** column name */
    SignaturesSelectColumn["Threshold"] = "threshold";
    /** column name */
    SignaturesSelectColumn["TransactionBlockHeight"] = "transaction_block_height";
    /** column name */
    SignaturesSelectColumn["TransactionVersion"] = "transaction_version";
    /** column name */
    SignaturesSelectColumn["Type"] = "type";
})(SignaturesSelectColumn || (SignaturesSelectColumn = {}));
/** select columns of table "table_items" */
export var TableItemsSelectColumn;
(function (TableItemsSelectColumn) {
    /** column name */
    TableItemsSelectColumn["DecodedKey"] = "decoded_key";
    /** column name */
    TableItemsSelectColumn["DecodedValue"] = "decoded_value";
    /** column name */
    TableItemsSelectColumn["Key"] = "key";
    /** column name */
    TableItemsSelectColumn["TableHandle"] = "table_handle";
    /** column name */
    TableItemsSelectColumn["TransactionVersion"] = "transaction_version";
    /** column name */
    TableItemsSelectColumn["WriteSetChangeIndex"] = "write_set_change_index";
})(TableItemsSelectColumn || (TableItemsSelectColumn = {}));
/** select columns of table "table_metadatas" */
export var TableMetadatasSelectColumn;
(function (TableMetadatasSelectColumn) {
    /** column name */
    TableMetadatasSelectColumn["Handle"] = "handle";
    /** column name */
    TableMetadatasSelectColumn["KeyType"] = "key_type";
    /** column name */
    TableMetadatasSelectColumn["ValueType"] = "value_type";
})(TableMetadatasSelectColumn || (TableMetadatasSelectColumn = {}));
/** select columns of table "legacy_migration_v1.token_activities" */
export var TokenActivitiesSelectColumn;
(function (TokenActivitiesSelectColumn) {
    /** column name */
    TokenActivitiesSelectColumn["CoinAmount"] = "coin_amount";
    /** column name */
    TokenActivitiesSelectColumn["CoinType"] = "coin_type";
    /** column name */
    TokenActivitiesSelectColumn["CollectionDataIdHash"] = "collection_data_id_hash";
    /** column name */
    TokenActivitiesSelectColumn["CollectionName"] = "collection_name";
    /** column name */
    TokenActivitiesSelectColumn["CreatorAddress"] = "creator_address";
    /** column name */
    TokenActivitiesSelectColumn["EventAccountAddress"] = "event_account_address";
    /** column name */
    TokenActivitiesSelectColumn["EventCreationNumber"] = "event_creation_number";
    /** column name */
    TokenActivitiesSelectColumn["EventIndex"] = "event_index";
    /** column name */
    TokenActivitiesSelectColumn["EventSequenceNumber"] = "event_sequence_number";
    /** column name */
    TokenActivitiesSelectColumn["FromAddress"] = "from_address";
    /** column name */
    TokenActivitiesSelectColumn["Name"] = "name";
    /** column name */
    TokenActivitiesSelectColumn["PropertyVersion"] = "property_version";
    /** column name */
    TokenActivitiesSelectColumn["ToAddress"] = "to_address";
    /** column name */
    TokenActivitiesSelectColumn["TokenAmount"] = "token_amount";
    /** column name */
    TokenActivitiesSelectColumn["TokenDataIdHash"] = "token_data_id_hash";
    /** column name */
    TokenActivitiesSelectColumn["TransactionTimestamp"] = "transaction_timestamp";
    /** column name */
    TokenActivitiesSelectColumn["TransactionVersion"] = "transaction_version";
    /** column name */
    TokenActivitiesSelectColumn["TransferType"] = "transfer_type";
})(TokenActivitiesSelectColumn || (TokenActivitiesSelectColumn = {}));
/** select columns of table "token_activities_v2" */
export var TokenActivitiesV2SelectColumn;
(function (TokenActivitiesV2SelectColumn) {
    /** column name */
    TokenActivitiesV2SelectColumn["AfterValue"] = "after_value";
    /** column name */
    TokenActivitiesV2SelectColumn["BeforeValue"] = "before_value";
    /** column name */
    TokenActivitiesV2SelectColumn["EntryFunctionIdStr"] = "entry_function_id_str";
    /** column name */
    TokenActivitiesV2SelectColumn["EventAccountAddress"] = "event_account_address";
    /** column name */
    TokenActivitiesV2SelectColumn["EventIndex"] = "event_index";
    /** column name */
    TokenActivitiesV2SelectColumn["FromAddress"] = "from_address";
    /** column name */
    TokenActivitiesV2SelectColumn["IsFungibleV2"] = "is_fungible_v2";
    /** column name */
    TokenActivitiesV2SelectColumn["PropertyVersionV1"] = "property_version_v1";
    /** column name */
    TokenActivitiesV2SelectColumn["ToAddress"] = "to_address";
    /** column name */
    TokenActivitiesV2SelectColumn["TokenAmount"] = "token_amount";
    /** column name */
    TokenActivitiesV2SelectColumn["TokenDataId"] = "token_data_id";
    /** column name */
    TokenActivitiesV2SelectColumn["TokenStandard"] = "token_standard";
    /** column name */
    TokenActivitiesV2SelectColumn["TransactionTimestamp"] = "transaction_timestamp";
    /** column name */
    TokenActivitiesV2SelectColumn["TransactionVersion"] = "transaction_version";
    /** column name */
    TokenActivitiesV2SelectColumn["Type"] = "type";
})(TokenActivitiesV2SelectColumn || (TokenActivitiesV2SelectColumn = {}));
/** select "token_activities_v2_aggregate_bool_exp_bool_and_arguments_columns" columns of table "token_activities_v2" */
export var TokenActivitiesV2SelectColumnTokenActivitiesV2AggregateBoolExpBoolAndArgumentsColumns;
(function (TokenActivitiesV2SelectColumnTokenActivitiesV2AggregateBoolExpBoolAndArgumentsColumns) {
    /** column name */
    TokenActivitiesV2SelectColumnTokenActivitiesV2AggregateBoolExpBoolAndArgumentsColumns["IsFungibleV2"] = "is_fungible_v2";
})(TokenActivitiesV2SelectColumnTokenActivitiesV2AggregateBoolExpBoolAndArgumentsColumns || (TokenActivitiesV2SelectColumnTokenActivitiesV2AggregateBoolExpBoolAndArgumentsColumns = {}));
/** select "token_activities_v2_aggregate_bool_exp_bool_or_arguments_columns" columns of table "token_activities_v2" */
export var TokenActivitiesV2SelectColumnTokenActivitiesV2AggregateBoolExpBoolOrArgumentsColumns;
(function (TokenActivitiesV2SelectColumnTokenActivitiesV2AggregateBoolExpBoolOrArgumentsColumns) {
    /** column name */
    TokenActivitiesV2SelectColumnTokenActivitiesV2AggregateBoolExpBoolOrArgumentsColumns["IsFungibleV2"] = "is_fungible_v2";
})(TokenActivitiesV2SelectColumnTokenActivitiesV2AggregateBoolExpBoolOrArgumentsColumns || (TokenActivitiesV2SelectColumnTokenActivitiesV2AggregateBoolExpBoolOrArgumentsColumns = {}));
/** select columns of table "legacy_migration_v1.token_datas" */
export var TokenDatasSelectColumn;
(function (TokenDatasSelectColumn) {
    /** column name */
    TokenDatasSelectColumn["CollectionDataIdHash"] = "collection_data_id_hash";
    /** column name */
    TokenDatasSelectColumn["CollectionName"] = "collection_name";
    /** column name */
    TokenDatasSelectColumn["CreatorAddress"] = "creator_address";
    /** column name */
    TokenDatasSelectColumn["DefaultProperties"] = "default_properties";
    /** column name */
    TokenDatasSelectColumn["Description"] = "description";
    /** column name */
    TokenDatasSelectColumn["DescriptionMutable"] = "description_mutable";
    /** column name */
    TokenDatasSelectColumn["LargestPropertyVersion"] = "largest_property_version";
    /** column name */
    TokenDatasSelectColumn["Maximum"] = "maximum";
    /** column name */
    TokenDatasSelectColumn["MaximumMutable"] = "maximum_mutable";
    /** column name */
    TokenDatasSelectColumn["MetadataUri"] = "metadata_uri";
    /** column name */
    TokenDatasSelectColumn["Name"] = "name";
    /** column name */
    TokenDatasSelectColumn["PayeeAddress"] = "payee_address";
    /** column name */
    TokenDatasSelectColumn["PropertiesMutable"] = "properties_mutable";
    /** column name */
    TokenDatasSelectColumn["RoyaltyMutable"] = "royalty_mutable";
    /** column name */
    TokenDatasSelectColumn["RoyaltyPointsDenominator"] = "royalty_points_denominator";
    /** column name */
    TokenDatasSelectColumn["RoyaltyPointsNumerator"] = "royalty_points_numerator";
    /** column name */
    TokenDatasSelectColumn["Supply"] = "supply";
    /** column name */
    TokenDatasSelectColumn["TokenDataIdHash"] = "token_data_id_hash";
    /** column name */
    TokenDatasSelectColumn["TransactionTimestamp"] = "transaction_timestamp";
    /** column name */
    TokenDatasSelectColumn["TransactionVersion"] = "transaction_version";
    /** column name */
    TokenDatasSelectColumn["UriMutable"] = "uri_mutable";
})(TokenDatasSelectColumn || (TokenDatasSelectColumn = {}));
/** select columns of table "legacy_migration_v1.token_ownerships" */
export var TokenOwnershipsSelectColumn;
(function (TokenOwnershipsSelectColumn) {
    /** column name */
    TokenOwnershipsSelectColumn["Amount"] = "amount";
    /** column name */
    TokenOwnershipsSelectColumn["CollectionDataIdHash"] = "collection_data_id_hash";
    /** column name */
    TokenOwnershipsSelectColumn["CollectionName"] = "collection_name";
    /** column name */
    TokenOwnershipsSelectColumn["CreatorAddress"] = "creator_address";
    /** column name */
    TokenOwnershipsSelectColumn["Name"] = "name";
    /** column name */
    TokenOwnershipsSelectColumn["OwnerAddress"] = "owner_address";
    /** column name */
    TokenOwnershipsSelectColumn["PropertyVersion"] = "property_version";
    /** column name */
    TokenOwnershipsSelectColumn["TableHandle"] = "table_handle";
    /** column name */
    TokenOwnershipsSelectColumn["TableType"] = "table_type";
    /** column name */
    TokenOwnershipsSelectColumn["TokenDataIdHash"] = "token_data_id_hash";
    /** column name */
    TokenOwnershipsSelectColumn["TransactionTimestamp"] = "transaction_timestamp";
    /** column name */
    TokenOwnershipsSelectColumn["TransactionVersion"] = "transaction_version";
})(TokenOwnershipsSelectColumn || (TokenOwnershipsSelectColumn = {}));
/** select columns of table "legacy_migration_v1.tokens" */
export var TokensSelectColumn;
(function (TokensSelectColumn) {
    /** column name */
    TokensSelectColumn["CollectionDataIdHash"] = "collection_data_id_hash";
    /** column name */
    TokensSelectColumn["CollectionName"] = "collection_name";
    /** column name */
    TokensSelectColumn["CreatorAddress"] = "creator_address";
    /** column name */
    TokensSelectColumn["Name"] = "name";
    /** column name */
    TokensSelectColumn["PropertyVersion"] = "property_version";
    /** column name */
    TokensSelectColumn["TokenDataIdHash"] = "token_data_id_hash";
    /** column name */
    TokensSelectColumn["TokenProperties"] = "token_properties";
    /** column name */
    TokensSelectColumn["TransactionTimestamp"] = "transaction_timestamp";
    /** column name */
    TokensSelectColumn["TransactionVersion"] = "transaction_version";
})(TokensSelectColumn || (TokensSelectColumn = {}));
/** select columns of table "user_transactions" */
export var UserTransactionsSelectColumn;
(function (UserTransactionsSelectColumn) {
    /** column name */
    UserTransactionsSelectColumn["BlockHeight"] = "block_height";
    /** column name */
    UserTransactionsSelectColumn["EntryFunctionContractAddress"] = "entry_function_contract_address";
    /** column name */
    UserTransactionsSelectColumn["EntryFunctionFunctionName"] = "entry_function_function_name";
    /** column name */
    UserTransactionsSelectColumn["EntryFunctionIdStr"] = "entry_function_id_str";
    /** column name */
    UserTransactionsSelectColumn["EntryFunctionModuleName"] = "entry_function_module_name";
    /** column name */
    UserTransactionsSelectColumn["Epoch"] = "epoch";
    /** column name */
    UserTransactionsSelectColumn["ExpirationTimestampSecs"] = "expiration_timestamp_secs";
    /** column name */
    UserTransactionsSelectColumn["GasUnitPrice"] = "gas_unit_price";
    /** column name */
    UserTransactionsSelectColumn["MaxGasAmount"] = "max_gas_amount";
    /** column name */
    UserTransactionsSelectColumn["ParentSignatureType"] = "parent_signature_type";
    /** column name */
    UserTransactionsSelectColumn["Sender"] = "sender";
    /** column name */
    UserTransactionsSelectColumn["SequenceNumber"] = "sequence_number";
    /** column name */
    UserTransactionsSelectColumn["Timestamp"] = "timestamp";
    /** column name */
    UserTransactionsSelectColumn["Version"] = "version";
})(UserTransactionsSelectColumn || (UserTransactionsSelectColumn = {}));
