// @aptos-labs/ts-sdk/dist/bcs/consts.js
// Copyright © Aptos Foundation
// SPDX-License-Identifier: Apache-2.0
// Upper bound values for uint8, uint16, uint64 etc.  These are all derived as
// 2^N - 1, where N is the number of bits in the type.
export const MAX_U8_NUMBER = 255;
export const MAX_U16_NUMBER = 65535;
export const MAX_U32_NUMBER = 4294967295;
export const MAX_U64_BIG_INT = 18446744073709551615n;
export const MAX_U128_BIG_INT = 340282366920938463463374607431768211455n;
export const MAX_U256_BIG_INT = 115792089237316195423570985008687907853269984665640564039457584007913129639935n;
// Signed integer bounds
export const MIN_I8_NUMBER = -128;
export const MAX_I8_NUMBER = 127;
export const MIN_I16_NUMBER = -32768;
export const MAX_I16_NUMBER = 32767;
export const MIN_I32_NUMBER = -2147483648;
export const MAX_I32_NUMBER = 2147483647;
export const MIN_I64_BIG_INT = -9223372036854775808n;
export const MAX_I64_BIG_INT = 9223372036854775807n;
export const MIN_I128_BIG_INT = -170141183460469231731687303715884105728n;
export const MAX_I128_BIG_INT = 170141183460469231731687303715884105727n;
export const MIN_I256_BIG_INT = -57896044618658097711785492504343953926634992332820282019728792003956564819968n;
export const MAX_I256_BIG_INT = 57896044618658097711785492504343953926634992332820282019728792003956564819967n;
