// @aptos-labs/aptos-cli/dist/utils/ghOperations.js
import { GH_CLI_DOWNLOAD_URL, PNAME } from "./consts.js";
const getGitHubHeaders = () => {
    const headers = {
        Accept: "application/vnd.github.v3+json",
        "User-Agent": "aptos-cli-npm",
    };
    const token = process.env.GITHUB_TOKEN;
    if (token) {
        headers.Authorization = `Bearer ${token}`;
    }
    return headers;
};
export const getUserSpecifiedVersion = () => {
    const version = process.env.APTOS_CLI_VERSION;
    if (!version) {
        return undefined;
    }
    const stripped = version.replace(/^v/, "");
    if (!/^\d+\.\d+\.\d+(-[a-zA-Z0-9.]+)?$/.test(stripped)) {
        throw new Error(`Invalid APTOS_CLI_VERSION: "${version}". Must match MAJOR.MINOR.PATCH (e.g. "4.5.0" or "v4.5.0").`);
    }
    return stripped;
};
export const hasUserSpecifiedVersion = () => {
    return !!process.env.APTOS_CLI_VERSION;
};
export const validateVersionExists = async (version, targetPlatform) => {
    const url = `${GH_CLI_DOWNLOAD_URL}/${PNAME}-v${version}/${PNAME}-${version}-${targetPlatform}.zip`;
    try {
        const response = await fetch(url, {
            method: "HEAD",
            headers: getGitHubHeaders(),
        });
        return response.ok;
    }
    catch {
        return false;
    }
};
export const getCliVersion = async (targetPlatform) => {
    const userVersion = getUserSpecifiedVersion();
    if (userVersion) {
        if (targetPlatform) {
            const exists = await validateVersionExists(userVersion, targetPlatform);
            if (!exists) {
                throw new Error(`Specified version ${userVersion} does not exist or is not available for ${targetPlatform}. ` +
                    `Check https://github.com/aptos-labs/aptos-core/releases for available versions.`);
            }
        }
        return userVersion;
    }
    return getLatestVersionGh();
};
export const getLatestVersionGh = async () => {
    const prefix = `${PNAME}-v`;
    const url = "https://api.github.com/repos/aptos-labs/aptos-core/releases?per_page=100";
    let response;
    try {
        response = await fetch(url, { headers: getGitHubHeaders() });
    }
    catch (error) {
        throw new Error(`Failed to fetch releases from GitHub: ${error instanceof Error ? error.message : String(error)}`);
    }
    if (!response.ok) {
        if (response.status === 403 || response.status === 429) {
            throw new Error("GitHub API rate limit exceeded. Please try again later or set a GITHUB_TOKEN environment variable.");
        }
        throw new Error(`GitHub API request failed with status ${response.status}: ${response.statusText}`);
    }
    let releases;
    try {
        releases = await response.json();
    }
    catch {
        throw new Error("Failed to parse GitHub API response as JSON");
    }
    if (!Array.isArray(releases)) {
        throw new Error("Unexpected response format from GitHub API");
    }
    for (const release of releases) {
        if (release.tag_name?.startsWith(prefix)) {
            return release.tag_name.replace(prefix, "");
        }
    }
    throw new Error("Could not determine latest version of Aptos CLI. No matching release found in the last 100 releases.");
};
export const getAssetDigest = async (version, assetName) => {
    const tag = `${PNAME}-v${version}`;
    const url = `https://api.github.com/repos/aptos-labs/aptos-core/releases/tags/${tag}`;
    try {
        const response = await fetch(url, { headers: getGitHubHeaders() });
        if (!response.ok) {
            return null;
        }
        const release = await response.json();
        const asset = release.assets?.find((a) => a.name === assetName);
        if (!asset?.digest) {
            return null;
        }
        const digest = asset.digest;
        if (digest.startsWith("sha256:")) {
            return digest.slice(7);
        }
        return null;
    }
    catch {
        return null;
    }
};
