// @aptos-labs/aptos-cli/dist/utils/windowsPackageManagers.js
import { execSync } from "node:child_process";
import { existsSync } from "node:fs";
import { executableIsAvailable } from "./executableIsAvailable.js";
export const isWingetAvailable = () => {
    return executableIsAvailable("winget");
};
export const isChocoAvailable = () => {
    return executableIsAvailable("choco");
};
export const isInstalledViaWinget = () => {
    if (!isWingetAvailable()) {
        return false;
    }
    try {
        const output = execSync("winget list --id Aptos.Aptos", {
            encoding: "utf8",
            stdio: ["pipe", "pipe", "pipe"],
        });
        return output.includes("Aptos.Aptos");
    }
    catch {
        return false;
    }
};
export const isInstalledViaChoco = () => {
    if (!isChocoAvailable()) {
        return false;
    }
    try {
        const output = execSync("choco list aptos", {
            encoding: "utf8",
            stdio: ["pipe", "pipe", "pipe"],
        });
        return output.toLowerCase().includes("aptos");
    }
    catch {
        return false;
    }
};
export const getCliPathWinget = () => {
    const localAppData = process.env.LOCALAPPDATA;
    const programFiles = process.env.PROGRAMFILES;
    const possiblePaths = [
        localAppData &&
            `${localAppData}\\Microsoft\\WinGet\\Packages\\Aptos.Aptos_Microsoft.Winget.Source_8wekyb3d8bbwe\\aptos.exe`,
        programFiles && `${programFiles}\\Aptos\\aptos.exe`,
        localAppData && `${localAppData}\\Aptos\\aptos.exe`,
    ];
    for (const path of possiblePaths) {
        if (path && existsSync(path)) {
            return path;
        }
    }
    try {
        const output = execSync("where aptos", {
            encoding: "utf8",
            stdio: ["pipe", "pipe", "pipe"],
        }).trim();
        if (output && existsSync(output.split("\n")[0])) {
            return output.split("\n")[0];
        }
    }
    catch {
    }
    return null;
};
export const getCliPathChoco = () => {
    const chocoPath = `${process.env.ChocolateyInstall || "C:\\ProgramData\\chocolatey"}\\bin\\aptos.exe`;
    if (existsSync(chocoPath)) {
        return chocoPath;
    }
    return null;
};
export const installViaWinget = () => {
    console.log("Installing Aptos CLI via winget...");
    execSync("winget install --id Aptos.Aptos --silent --accept-package-agreements --accept-source-agreements", {
        stdio: "inherit",
    });
};
export const installViaChoco = () => {
    console.log("Installing Aptos CLI via Chocolatey...");
    execSync("choco install aptos -y", { stdio: "inherit" });
};
export const updateViaWinget = () => {
    console.log("Updating Aptos CLI via winget...");
    execSync("winget upgrade --id Aptos.Aptos --silent --accept-package-agreements --accept-source-agreements", {
        stdio: "inherit",
    });
};
export const updateViaChoco = () => {
    console.log("Updating Aptos CLI via Chocolatey...");
    execSync("choco upgrade aptos -y", { stdio: "inherit" });
};
export const detectWindowsPackageManager = () => {
    if (isInstalledViaWinget()) {
        return "winget";
    }
    if (isInstalledViaChoco()) {
        return "choco";
    }
    return null;
};
