// @aptos-labs/ts-sdk/dist/types/ans.js
/**
 * Policy for determining how subdomains expire in relation to their parent domain.
 */
export var SubdomainExpirationPolicy;
(function (SubdomainExpirationPolicy) {
    /**
     * The subdomain will expire independently of the domain. The owner of the
     * domain can manually set when the subdomain expires.
     */
    SubdomainExpirationPolicy[SubdomainExpirationPolicy["Independent"] = 0] = "Independent";
    /**
     * The subdomain will expire at the same time as the domain.
     */
    SubdomainExpirationPolicy[SubdomainExpirationPolicy["FollowsDomain"] = 1] = "FollowsDomain";
})(SubdomainExpirationPolicy || (SubdomainExpirationPolicy = {}));
/**
 * The status of the name.
 */
export var ExpirationStatus;
(function (ExpirationStatus) {
    /**
     * The name no longer functions as a primary or target name. It is open to
     * being claimed by the public.
     */
    ExpirationStatus["Expired"] = "expired";
    /**
     * The name is past it's expiration date, but only claimable by the current
     * owner of the name. It does not function as a primary or target name.
     */
    ExpirationStatus["InGracePeriod"] = "in_grace_period";
    /**
     * The name is in good standing.
     */
    ExpirationStatus["Active"] = "active";
})(ExpirationStatus || (ExpirationStatus = {}));
